src/lib/Zikula/Core/Controller/AbstractBlockController.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Zikula\Core\Controller;
+
+abstract class AbstractBlockController extends AbstractController
+{
+    /**
+     * Initialise interface.
+     *
+     * @return void
+     */
+    abstract public function init();
+
+    /**
+     * Get info interface.
+     *
+     * @return array BlockInfo.
+     */
+    abstract public function info();
+
+    /**
+     * Display block.
+     *
+     * @param array $blockInfo Blockinfo.
+     *
+     * @return array Blockinfo.
+     */
+    abstract public function display($blockInfo);
+
+    /**
+     * Modify block interface.
+     *
+     * @param array $blockInfo Block info.
+     *
+     * @return string
+     */
+    public function modify($blockInfo)
+    {
+        return '';
+    }
+
+    /**
+     * Update block interface.
+     *
+     * @param array $blockInfo Block info.
+     *
+     * @return array Blockinfo.
+     */
+    public function update($blockInfo)
+    {
+        return $blockInfo;
+    }
+
+    /**
+     * Magic method to for method_not_found events.
+     *
+     * @param string $method Method invoked.
+     * @param array  $args   Arguments.
+     *
+     * @throws \BadMethodCallException If no event responds.
+     *
+     * @return string Data.
+     */
+    public function __call($method, $args)
+    {
+        $event = new \Zikula\Core\Event\GenericEvent($this, array('method' => $method, 'args' => $args));
+        $this->get('eventmanager')->dispatch('block.method_not_found', $event);
```

Comment:
The event name [is a 1.3 (legacy) name](https://github.com/zikula/core/blob/1.4/src/lib/legacy/Zikula/Controller/AbstractBlock.php#L81)

how about removing the magic method altogether?